### PR TITLE
migrate: vz_line_chart2

### PR DESCRIPTION
**Pull request by stephanwlee**

### CODE REVIEW

File: tensorboard/components_polymer3/vz_line_chart2/log-scale.ts [hunk 1]
```diff
@@ -12,170 +12,145 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-namespace vz_line_chart2 {
-  // Smallest positive non-zero value represented by IEEE 754 binary (64 bit)
-  // floating-point number.
-  // https://www.ecma-international.org/ecma-262/5.1/#sec-8.5
```
Comment (psybuzz) [inline tensorboard/components_polymer3/vz_line_chart2/log-scale.ts:18]: Can we preserve this comment?

File: tensorboard/components_polymer3/vz_line_chart2/line-chart.ts [hunk 2]
```diff
@@ -12,1043 +12,945 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+  LINEAR = 'linear',
+}
+export type LineChartStatus = {
+  smoothingEnabled: boolean;
+};
+
+/**
+ * Adds private APIs for default swatch column on the first column.
+ */
+export interface LineChartTooltipColumn extends TooltipColumn {
```
Comment (psybuzz) [inline tensorboard/components_polymer3/vz_line_chart2/line-chart.ts:54]: This seems to be used only in this file.  No need to `export`?

File: tensorboard/components_polymer3/vz_line_chart2/line-chart.ts [hunk 3]
```diff
@@ -12,1043 +12,945 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-namespace vz_line_chart2 {
-  /**
-   * An interface that describes a fill area to visualize. The fill area is
-   * visualized with a less intense version of the color for a given series.
-   */
```
Comment (psybuzz) [inline tensorboard/components_polymer3/vz_line_chart2/line-chart.ts:19]: Preserve comment?

File: tensorboard/components_polymer3/vz_line_chart2/vz-line-chart2.ts [hunk 4]
```diff
@@ -12,473 +12,498 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
-  Polymer({
-    is: 'vz-line-chart2',
-    properties: {
-      /**
-       * Scale that maps series names to colors. The default colors are from
-       * d3.schemeCategory10. Use this property to replace the default line
-       * colors with colors of your own choice.
-       * @type {Plottable.Scales.Color}
-       * @required
-       */
```
Comment (psybuzz) [inline tensorboard/components_polymer3/vz_line_chart2/vz-line-chart2.ts:67]: We should probably preserve comments on all these properties, as well.

File: tensorboard/components_polymer3/vz_line_chart2/vz-line-chart2.ts [hunk 5]
```diff
@@ -139,58 +144,157 @@ class VzLineChart2 extends LegacyElementMixin(PolymerElement) {
...
 
+  /**
+   * This is a helper field for automatically generating commonly used
+   * functions for xComponentsCreationMethod. Valid values are what can
+   * be processed by vz_chart_helpers.getXComponents() and include
+   * "step", "wall_time", and "relative".
+   */
   @property({type: String})
-  xType: string = '';
+  xType: XType = XType.STEP;
```
Comment (psybuzz) [inline tensorboard/components_polymer3/vz_line_chart2/vz-line-chart2.ts:194]: Before, a falsy default tells `_makeChart` to use `xComponentsCreationMethod` passed in.
By having a truthy default, will this change the behavior and always override the `xComponentsCreationMethod`?

Comment (stephanwlee) [reply tensorboard/components_polymer3/vz_line_chart2/vz-line-chart2.ts]: Changed it back to `null` for falsey.

**Pull request merged**